### ISSUE TITLE
🐛 BUG: vitest watch mode does not work for changes in non-test files

### ISSUE THREAD

**Issue by mhujer** [cloudflare/workers-sdk#2600]
### Which Cloudflare product(s) does this pertain to?

Wrangler

### What version of `Wrangler` are you using?

2.8.1

### What operating system are you using?

Windows

### Describe the Bug

`vitest` watch mode does not work for changes in the worker files, only for the changes in test files.

## Steps to reproduce

1. create a new project with Wrangler (with TS and tests)
2. run `npm test` - vitest starts in watch mode
3. add extra `!` to the expected message in [`index.test.ts`](https://github.com/cloudflare/wrangler2/blob/416babf050ed3608e0fd747111561a4c7207185e/packages/wrangler/templates/init-tests/test-vitest-new-worker.ts#L22) (`expect(text).toMatchInlineSnapshot("Hello World!!");`), the vitest notices this change and reruns the tests which now fail
4. fix the code in [`index.ts`](https://github.com/cloudflare/wrangler2/blob/416babf050ed3608e0fd747111561a4c7207185e/packages/wrangler/templates/new-worker.ts#L28) to also include the extra `!`, the tests are not re-ran

## Expected outcome
Tests are re-ran even after changes in the source code, not just after changes to the tests

## Workaround
I guess that's because vitest has some smart way of determining the dependencies of the tests, so it watches just the specific files and `worker = await unstable_dev("src/index.ts"` is not enough to let it know that `src/index.ts` should be also watched.

If I add `import * as workerFile from './index';` and `console.log(workerFile);` to the `index.test.ts` so the dependency between files is defined in a JS-way, test are re-ran also after changes to `index.ts`

**Comment by mrbbot** [comment]
Hey! You should be able to use [`forceRerunTriggers`](https://vitest.dev/config/#forcereruntriggers) for this. Create a `vitest.config.ts` file containing the following:

```ts
import { configDefaults, defineConfig } from "vitest/config";

export default defineConfig({
  test: {
    forceRerunTriggers: [...configDefaults.forceRerunTriggers, "src/index.ts"],
  },
});
```

**Comment by mrbbot** [comment]
...admittedly we should add this for you with `wrangler init`, so I'll re-open to track that. 😅

**Comment by mhujer** [comment]
@mrbbot Thank for the answer! I tried the approach you suggested, but it seems that adding `src/index.ts` to `forceRerunTriggers` is not enough. In my testing it works if all the worker code is located in `src/index.ts`, but ignores changes in other included files, so probably the vitest dependencies graph resolution is not applied to it.

```ts
// generateResponse.ts
export function generateResponse() {
    return new Response("Hello World!!"); // <--- changes to this line are ignored.
}

//index.ts
import { generateResponse } from "./generateResponse";

export default {
	async fetch(
		request: Request,
		env: Env,
		ctx: ExecutionContext
	): Promise<Response> {
		return generateResponse();
	},
};
```

Therefor fix for the default project generated with `wrangler init` should be different -maybe something like `src/*.ts` in `forceRerunTriggers`. 

For my setup I simplified the workaround just to adding `import './index';` to the top of the `index.test.ts`. This way the vitest dependencies resolution works fine for changes in other files than `index.ts`.
